runtime/src/main/kotlin/kotlin/native/concurrent/Freezing.kt

```diff
@@ -42,12 +46,37 @@ public fun <T> T.freeze(): T {
 public val Any?.isFrozen
     get() = isFrozenInternal(this)
 
+
+/**
+ * Transforms an object to the frozen form by creating frozen version of the object, by
+ * deep copy of mutable objects and reusing frozen references. Note, that consequent calls
+ * of this function may return different instances. To make a clone class must be eitehr data class or
+ * explicitly marked by [kotlin.ShareByValue] annotation.
+ *
+ * @return the frozen object form
+ * @see ensureNeverFrozen
+ * @see ensureNeverCloned
+ * @see kotlin.ShareByValue
+ */
+public inline fun <reified T> T.toFrozen(): T =
```

Comment:
Have you considered naming this something like `copyFrozen()`, since users are used to the `copy()` function in data classes? I think it also conveys better how this is different from `freeze()`.